lib/active_fulfillment/services/amazon_mws.rb

```diff
@@ -15,6 +15,7 @@ class AmazonMarketplaceWebService < Service
     VERSION = "2010-10-01"
 
     SUCCESS, FAILURE, ERROR = 'Accepted', 'Failure', 'Error'
+    XML_FAILURE_RESPONSE = { :success => FAILURE }.freeze
```

Comment:
Not for hashes and array. They get re-allocated if we freeze them in a inlined way.

```
 ~  irb
2.3.0 :001 > def foo
2.3.0 :002?>   a = { test: '123' }.freeze
2.3.0 :003?>   puts a.object_id
2.3.0 :004?> end
 => :foo
2.3.0 :005 > foo
70170052690800
 => nil
2.3.0 :006 > foo
70170052678260
 => nil
```